src/libraries/System.Private.CoreLib/src/System/Half.cs

```diff
@@ -607,30 +608,33 @@ public bool TryFormat(Span<char> destination, out int charsWritten, [StringSynta
         /// <returns><paramref name="value" /> converted to its nearest representable half-precision floating-point value.</returns>
         public static explicit operator Half(float value)
         {
-            const int SingleMaxExponent = 0xFF;
-
-            uint floatInt = BitConverter.SingleToUInt32Bits(value);
-            bool sign = (floatInt & float.SignMask) >> float.SignShift != 0;
-            int exp = (int)(floatInt & float.BiasedExponentMask) >> float.BiasedExponentShift;
-            uint sig = floatInt & float.TrailingSignificandMask;
-
-            if (exp == SingleMaxExponent)
-            {
-                if (sig != 0) // NaN
-                {
-                    return CreateHalfNaN(sign, (ulong)sig << 41); // Shift the significand bits to the left end
-                }
-                return sign ? NegativeInfinity : PositiveInfinity;
-            }
-
-            uint sigHalf = sig >> 9 | ((sig & 0x1FFU) != 0 ? 1U : 0U); // RightShiftJam
-
-            if ((exp | (int)sigHalf) == 0)
-            {
-                return new Half(sign, 0, 0);
-            }
-
-            return new Half(RoundPackToHalf(sign, (short)(exp - 0x71), (ushort)(sigHalf | 0x4000)));
+            Vector128<uint> v0 = Vector128.CreateScalarUnsafe(0x3880_0000u); //Minimum exponent for rounding
+            Vector128<uint> v1 = Vector128.CreateScalarUnsafe(0x3800_0000u); //Exponent displacement #1
+            Vector128<uint> v3 = Vector128.CreateScalarUnsafe(0x7f80_0000u); //Exponent mask
+            Vector128<uint> v4 = Vector128.CreateScalarUnsafe(0x0680_0000u); //Exponent displacement #2
+            Vector128<float> v5 = Vector128.CreateScalarUnsafe(65520.0f);     //Maximum value that is not Infinity in Half
```

Comment:
We can't use `Vector128` without `IsHardwareAcceelerated` checks. That will regress various platforms.

This either needs to be purely scalar logic and avoid the intrinsics entirely or there needs to be a fallback back for platforms without.

This is a case where the need to maintain a "semi-vectorized" and a "fallback" path is likely not worthwhile, so optimizing this to be purely scalar would be much preferred

Reply:
-- Purely scalar as in not using `Vector128` at all

Reply:
Optimization of `float.Min` helped it a lot. It now seems to emit `vminps` if one of two arguments is a constant.
`Vector128` is no longer present in the new code.
I hope the line:
```csharp
value = BitConverter.UInt32BitsToSingle(BitConverter.SingleToUInt32Bits(value) - Exponent112);
```
Would get optimized to `vpsubd`.